### PR TITLE
Multibarrierwell #5464

### LINKED ISSUE #5464: Well only update for MultiBarrierFunctionMaterial
We need to update MultiBarrierFunctionMaterial so that it has the same capability as BarrierFunctionMaterial to only add a wall to enforce eta ranges without adding a barrier if desired.

### PR DESCRIPTION
**Pull request by jbair34**
Rebased and ready to go. Closes #5464

### CODE REVIEW

File: modules/phase_field/src/materials/CrossTermBarrierFunctionMaterial.C [hunk 1]
```diff
@@ -30,17 +30,20 @@ CrossTermBarrierFunctionMaterial::CrossTermBarrierFunctionMaterial(const InputPa
     _prop_d2g(_num_eta)
 
 {
+  //loop counters
+  unsigned int i,j;
+
```
Comment (jwpeterson) [inline modules/phase_field/src/materials/CrossTermBarrierFunctionMaterial.C]: This is probably left over from rebasing, you need to remove these loop counters which are declared outside the loop.

File: modules/phase_field/src/materials/CrossTermBarrierFunctionMaterial.C [hunk 2]
```diff
@@ -70,6 +73,7 @@ CrossTermBarrierFunctionMaterial::CrossTermBarrierFunctionMaterial(const InputPa
 void
 CrossTermBarrierFunctionMaterial::computeQpProperties()
 {
+  unsigned int i,j;
```
Comment (jwpeterson) [inline modules/phase_field/src/materials/CrossTermBarrierFunctionMaterial.C]: same here

File: modules/phase_field/src/materials/MultiBarrierFunctionMaterial.C [hunk 3]
```diff
@@ -45,6 +47,14 @@ MultiBarrierFunctionMaterial::computeQpProperties()
   for (unsigned int i = 0; i < _num_eta; ++i)
   {
     const Real n = (*_eta[i])[_qp];
+
+    if (_well_only && n >= 0.0 && n <= 1.0) {
```
Comment (jwpeterson) [inline modules/phase_field/src/materials/MultiBarrierFunctionMaterial.C]: Curly brace goes on next line.

Comment (jbair34) [inline modules/phase_field/src/materials/MultiBarrierFunctionMaterial.C]: I will work on it tomorrow... I just went home for the day.

**Comment by jbair34** [comment]
About that curly bracket I was again copying the format of another material... the BarrierFunctionMaterial. Do you want me to change that one too?

**Comment by jwpeterson** [comment]
> About that curly bracket I was again copying the format of another material... the BarrierFunctionMaterial. Do you want me to change that one too?

Nah, we'll take care of it.

**Pull request merged**